torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -376,17 +377,23 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
   runPreAutodiffPassPipeline(copy);
 
   if (needsGradientInProfilingMode(copy->block())) {
+    RemoveProfileNodesAndSpecializeTypes(copy);
```

Comment:
I had the following in mind:
- CreateAutodiffSubgraphs
- Specialize the inputs of DifferentiableGraphs
- Add a guard based on the input requiresGrad.

My thinking was that the fallback function doesn't have a differentiable graph but only the nodes that have been moved inside the graph so it would be good to use as is.